vulture/config.py

```diff
@@ -0,0 +1,214 @@
+    incorrect_types = given_types - expected_types
+    if incorrect_types:
+        keys = ", ".join(sorted(item[0] for item in incorrect_types))
+        msg = f"Incorrect data-types for the config values: {keys}"
+        sys.exit(msg)
+
+
+def from_dict(data):
+    """
+    Create a new config dictionary from an existing one, assign possible
+    defaults and warn about unprocessed options.
+    """
+    _check_config_types(data)
+
+    # keep a copy of the keys, so we can keep track of any unprocessed
+    # values.
+    unknown_keys = set(data) - set(DEFAULTS)
+    if unknown_keys:
+        print(
+            f"Unknown configuration keys: {sorted(unknown_keys)}",
+            file=sys.stderr,
+        )
+        sys.exit("invalid Config")
+
+    output = {}
+    for key, default in DEFAULTS.items():
+        output[key] = data.get(key, default)
+    return output
+
+
+def _parse_toml(infile):
+    """
+    Parse a TOML file for config values.
+
+    It will search for a section named ``[tool.vulture]`` which contains the
+    same keys as the CLI arguments seen with ``--help``. All leading dashes are
+    removed and other dashes are replaced by underscores (so ``--sort-by-size``
+    becomes ``sort_by_size``).
+
+    Arguments containing multiple values are standard TOML lists.
+
+    Example::
+
+        [tool.vulture]
+        exclude = ["file*.py", "dir/"]
+        ignore_decorators = ["deco1", "deco2"]
+        ignore_names = ["name1", "name2"]
+        make_whitelist = true
+        min_confidence = 10
+        sort_by_size = true
+        verbose = true
+        paths = ["path1", "path2"]
+    """
+    data = toml.load(infile)
+    vulture_settings = data.get("tool", {}).get("vulture", {})
+    return from_dict(vulture_settings)
+
+
+def _parse_args(args=None):
+    """
+    Parse CLI arguments
+
+    :param args: A list of strings representing the CLI arguments. If left to
+        the default, this will default to ``sys.argv``.
+    """
+
+    def csv(exclude):
+        return exclude.split(",")
+
+    usage = "%(prog)s [options] PATH [PATH ...]"
+    version = f"vulture {__version__}"
+    glob_help = "Patterns may contain glob wildcards (*, ?, [abc], [!abc])."
+    parser = argparse.ArgumentParser(prog="vulture", usage=usage)
+    parser.add_argument(
+        "paths",
+        nargs="+",
+        metavar="PATH",
+        help="Paths may be Python files or directories. For each directory"
+        " Vulture analyzes all contained *.py files.",
+    )
+    parser.add_argument(
+        "--exclude",
+        metavar="PATTERNS",
+        type=csv,
+        default=[],
+        help="Comma-separated list of paths to ignore (e.g.,"
+        ' "*settings.py,docs/*.py"). {glob_help} A PATTERN without glob'
+        " wildcards is treated as *PATTERN*.".format(**locals()),
+    )
+    parser.add_argument(
+        "--ignore-decorators",
+        metavar="PATTERNS",
+        type=csv,
+        default=[],
```

Comment:
It would be good if we accessed DEFAULTS for retrieving the defaults everywhere.

Reply:
Good..... point.... :smile: 

Sone in 3ba6473